iceoryx_dust/container/include/iox/fixed_position_container.hpp

```diff
@@ -324,6 +324,9 @@ class FixedPositionContainer final
     IndexType m_size{0};
     IndexType m_begin_free{Index::FIRST};
     IndexType m_begin_used{Index::INVALID};
+
+    template <typename RhsType>
+    void init(RhsType&& rhs) noexcept;
```

Comment:
The implementation looks sound. Now it's up to the tests to find out if there are still some corner cases :)

btw, you cannot know this but we usually have the members at the very bottom. Could you please move it up. Something between line 179 and 320. Oh, and can you find a name that better describe what the function does? If you need suggestions feel free to ask :)

Reply:
Well, nothing comes to mind. Do you have any suggestions for a name :>

Reply:
Well, you know that naming is one of the two big problems in computer science. The other ones are cache coherency and off-by-1 errors :) 

How about `copy_and_move_impl`?

Reply:
Hahaha, that's true. I'll take it:)